apis/sgv2-restapi/src/main/java/io/stargate/sgv2/restapi/service/resources/schemas/Sgv2IndexesResourceImpl.java

```diff
@@ -80,22 +80,34 @@ public Uni<RestResponse<Map<String, Object>>> addIndex(
   @Override
   public Uni<RestResponse<Void>> deleteIndex(
       String keyspaceName, String tableName, String indexName, boolean ifExists) {
-    return queryWithTableAsync(
-            keyspaceName,
-            tableName,
-            true,
-            table -> {
-              if (!ifExists
-                  && table.getIndexesList().stream()
-                      .noneMatch(i -> indexName.equals(i.getName()))) {
-                throw new WebApplicationException(
-                    String.format("Index '%s' not found.", indexName), Status.NOT_FOUND);
+    final Query dropQuery =
+        new QueryBuilder().drop().index(keyspaceName, indexName).ifExists(ifExists).build();
+    if (ifExists) {
+      return executeQueryAsync(dropQuery).map(any -> RestResponse.noContent());
+    }
+
+    // Use direct query for checking existence
+    final Query existenceQuery =
```

Comment:
Same query as for GET INDEX except all we care about is one specific index. So select "kind" field.

Reply:
I am not understanding this. Why in case `ifExsist=false` we can not go through optimistic path?

I am also not understanding why we need to check the index first? and how is index not in the `CqlTable` schema.. Going for `system_schema` seems wrong imo.. Keyspace can not be prefixed with tenant and this will fail in some cases?

Reply:
So, why check? That's the way code was constructed when I started working: but I _think_ it was meant to give better user experience: perhaps deletion would simply succeed? I don't know the original reasoning.
Perhaps it'd be better simply to try to execute the query and see what happens.

Checking `system_schema` is used for other endpoints of Index access which is why I thought going that route would work (as it seems to).

But I can also change code not to do any validation and try to get failure from persistence backend.

The big question is indeed why `CqlTable` does not have/get updated information. Is it possible index information would not be part of hash calculation?

Reply:
@ivansenic I rewrote handling again to use Optimistic Query path but removing explicit validation by caller -- persistence backend will do that anyway, reporting it as expected. This works and gives expected fails on missing index. I think this makes sense even if we figure out the issue with `CqlTable` not containing index information.